common/http/src/main/java/io/helidon/common/http/DataChunk.java

```diff
@@ -194,4 +225,29 @@ default void release() {
     default boolean flush() {
         return false;
     }
+
+    /**
+     * Makes a copy of this data chunk including its underlying {@link ByteBuffer}. This
+     * may be necessary for caching in case {@link ByteBuffer#rewind()} is called to
+     * reuse a byte buffer. Note that only the actual bytes used in the data chunk are
+     * copied, the resulting data chunk's capacity may be less than the original.
+     *
+     * @return A copy of this data chunk.
+     */
+    default DataChunk duplicate() {
+        byte[] bytes = new byte[data().limit()];
+        DataChunk dup = DataChunk.create(data().get(bytes));
+        dup.data().position(0);
+        return dup;
+    }
+
+    /**
+     * Returns {@code true} if the underlying byte buffer of this chunk is read
+     * only or {@code false} otherwise.
+     *
+     * @return Immutability outcome.
+     */
+    default boolean isImmutable() {
```

Comment:
How about calling the flag readOnly instead ?

Reply:
Sure, we can use read only. But for consistency with other methods in that class, it should probably be named `isReadOnly`.

Reply:
Right that's what I meant, sorry for the confusion.